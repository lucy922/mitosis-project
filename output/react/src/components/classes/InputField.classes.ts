import  { IinputFieldClasses } from '../InputField'
export default <IinputFieldClasses>{
 base: "border-solid border-2 border-sky-300 outline-none w-96 rounded-md p-2"
}