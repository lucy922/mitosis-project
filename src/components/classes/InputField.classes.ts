import  { IinputFieldClasses } from '../InputField.lite'
export default <IinputFieldClasses>{
 base: "border-solid border-2 border-sky-300 outline-none w-96 rounded-md p-2"
}