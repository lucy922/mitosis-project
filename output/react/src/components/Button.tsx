import * as React from "react";

export interface IButtonClasses {
  base: string;
  variant: {
    default: string;
    primary: string;
    secondary: string;
  };
  size: {
    sm: string;
    md: string;
    lg: string;
  };
}
export interface IButtonProps {
  children?: string;
  variant?: keyof IButtonClasses["variant"];
  size?: keyof IButtonClasses["size"];
  disabled?: boolean;
  name?: string;
  type?: "button" | "reset" | "submit";
  onClick?: () => void;
}

import classes from "./classes/Button.classes";
import classNames from "classnames";

export default function Button(props: IButtonProps) {
  return (
    <div>
      <button
        data-testid="button"
        type="submit"
        disabled={props.disabled}
        onClick={(event) => props.onClick && props.onClick()}
        className={classNames(
          classes.base,
          classes.variant[props.variant ?? "default"],
          classes.size[props.size ?? "sm"]
        )}
      >
        {props.children}
      </button>
    </div>
  );
}
