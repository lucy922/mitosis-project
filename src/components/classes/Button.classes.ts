import { IButtonClasses } from "../Button.lite";

export default <IButtonClasses> {
  base: "p-2 rounded text-center text-white",
  variant: {
    default: "bg-green-800",
    primary: "bg-blue-800" ,
    secondary: 'bg-red-800'
  },
  size: {
    sm: "w-24",
    md: " w-32",
    lg: "w-64"
  }
  }