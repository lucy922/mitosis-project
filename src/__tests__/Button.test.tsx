import Button, { IButtonProps } from "../../output/react/src/components/Button";
import { create } from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";
import classNames from "classnames";
import classes from "../../output/react/src/components/classes/Button.classes";

describe("Button component", () => {
  it("should render button correctly", () => {
    render(<Button>Login</Button>);
    expect(screen.getByTestId("button")).toBeInTheDocument();
  });

  it("should fire onClick event", () => {
    const onClick = jest.fn();
    render(
      <Button onClick={onClick} type="submit">
        Login
      </Button>
    );
    const button = screen.getByTestId("button");
    fireEvent.click(button);
    expect(onClick).toHaveBeenCalled();
  });

  it("Button text", () => {
    render(<Button>Login</Button>);
    expect(screen.getByTestId("button")).toHaveTextContent("Login");
  });

  it("should be enabled", () => {
    render(<Button>Login</Button>);
    expect(screen.getByTestId("button")).not.toHaveAttribute("disabled");
  });

  it("should be disabled", () => {
    render(<Button disabled={true}>Login</Button>);
    expect(screen.getByTestId("button")).toBeDisabled();
  });

  it("Matches snapshot", () => {
    const tree = create(<Button>Login</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render primary button", () => {
    const variantProps: IButtonProps = {
      variant: "primary",
    };
    render(<Button {...variantProps}>Login</Button>);
    expect(screen.getByTestId("button")).toHaveClass(
      classNames(classes.variant.primary)
    );
    const tree = create(<Button {...variantProps}>primary</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render secondary button", () => {
    const variantProps: IButtonProps = {
      variant: "secondary",
    };
    render(<Button {...variantProps}>Login</Button>);
    expect(screen.getByTestId("button")).toHaveClass(
      classNames(classes.variant.secondary)
    );
    const tree = create(<Button {...variantProps}>secondary</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render size sm", () => {
    const sizeProps: IButtonProps = {
      size: "sm",
    };
    render(<Button {...sizeProps}>Login</Button>);
    expect(screen.getByTestId("button")).toHaveClass(
      classNames(classes.size.sm)
    );
    const tree = create(<Button {...sizeProps}>small</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render size md", () => {
    const sizeProps: IButtonProps = {
      size: "md",
    };
    render(<Button {...sizeProps}>Login</Button>);
    expect(screen.getByTestId("button")).toHaveClass(
      classNames(classes.size.md)
    );
    const tree = create(<Button {...sizeProps}>medium</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render size lg", () => {
    const sizeProps: IButtonProps = {
      size: "lg",
    };
    render(<Button {...sizeProps}>Login</Button>);
    expect(screen.getByTestId("button")).toHaveClass(
      classNames(classes.size.lg)
    );
    const tree = create(<Button {...sizeProps}>large</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
