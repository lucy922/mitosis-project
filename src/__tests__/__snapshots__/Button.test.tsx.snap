// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches snapshot 1`] = `
<div>
  <button
    className="p-2 rounded text-center text-white bg-green-800 w-24"
    data-testid="button"
    onClick={[Function]}
    type="submit"
  >
    Login
  </button>
</div>
`;

exports[`Button component should render primary button 1`] = `
<div>
  <button
    className="p-2 rounded text-center text-white bg-blue-800 w-24"
    data-testid="button"
    onClick={[Function]}
    type="submit"
  >
    primary
  </button>
</div>
`;

exports[`Button component should render secondary button 1`] = `
<div>
  <button
    className="p-2 rounded text-center text-white bg-red-800 w-24"
    data-testid="button"
    onClick={[Function]}
    type="submit"
  >
    secondary
  </button>
</div>
`;

exports[`Button component should render size lg 1`] = `
<div>
  <button
    className="p-2 rounded text-center text-white bg-green-800 w-64"
    data-testid="button"
    onClick={[Function]}
    type="submit"
  >
    large
  </button>
</div>
`;

exports[`Button component should render size md 1`] = `
<div>
  <button
    className="p-2 rounded text-center text-white bg-green-800  w-32"
    data-testid="button"
    onClick={[Function]}
    type="submit"
  >
    medium
  </button>
</div>
`;

exports[`Button component should render size sm 1`] = `
<div>
  <button
    className="p-2 rounded text-center text-white bg-green-800 w-24"
    data-testid="button"
    onClick={[Function]}
    type="submit"
  >
    small
  </button>
</div>
`;
